service/resource/s3objectv1/resource.go

```diff
@@ -17,28 +17,34 @@ const (
 // Config represents the configuration used to create a new cloudformation resource.
 type Config struct {
 	// Dependencies.
-	AwsService AwsService
-	Clients    Clients
-	Logger     micrologger.Logger
+	AwsService  AwsService
+	CertWatcher CertWatcher
+	Clients     Clients
+	CloudConfig CloudConfigService
+	Logger      micrologger.Logger
```

Comment:
Doesn't this also need the KeyWatcher from `randomkeytpr`?

Reply:
Not yet, the encoded `clusterKeys` generated by using the keys obtained by `KeyWatcher` in legacy are used to generate the master's cloudconfig https://github.com/giantswarm/aws-operator/blob/master/service/resource/legacyv1/launch_configuration.go#L45 but not for worker's https://github.com/giantswarm/aws-operator/blob/master/service/resource/legacyv1/launch_configuration.go#L50, once this is ported we'll need a `KeyWatcher` dependency too.